sdks/python/apache_beam/testing/util.py

```diff
@@ -154,41 +154,48 @@ def equal_to_per_window(expected_window_to_elements):
 # Note that equal_to checks if expected and actual are permutations of each
 # other. However, only permutations of the top level are checked. Therefore
 # [1,2] and [2,1] are considered equal and [[1,2]] and [[2,1]] are not.
-def equal_to(expected):
+def equal_to(expected, equals_fn=None):
 
-  def _equal(actual):
+  def _equal(actual, equals_fn=equals_fn):
     expected_list = list(expected)
 
     # Try to compare actual and expected by sorting. This fails with a
     # TypeError in Python 3 if different types are present in the same
     # collection. It can also raise false negatives for types that don't have
     # a deterministic sort order, like pyarrow Tables as of 0.14.1
-    try:
-      sorted_expected = sorted(expected)
-      sorted_actual = sorted(actual)
-      if sorted_expected != sorted_actual:
-        raise BeamAssertException(
-            'Failed assert: %r == %r' % (sorted_expected, sorted_actual))
+    if not equals_fn:
+      try:
+        sorted_expected = sorted(expected)
+        sorted_actual = sorted(actual)
+        if sorted_expected == sorted_actual:
+          return
+      except (BeamAssertException, TypeError):
```

Comment:
We can remove `BeamAssertException` from here. The remaining code in the block would not raise that exception.